src/utilities/core-utils.test.ts

```diff
@@ -91,6 +91,145 @@ describe("CoreUtils", () => {
+            expect(result).toMatchObject({});
+            expect(result).toBeEmpty();
+        });
+    });
+
+    // #endregion numericEnumToPojo
+
+    // -------------------------------------------------------------------------------------------------
+    // #region ObjectToArray
+    // -------------------------------------------------------------------------------------------------
+
+    describe("objectToArray", () => {
+        test("convert object to array of values", () => {
+            // Arrange
+            const testObject = {
+                name: "Jane",
+                age:  30
+            }
+
+            // Act
+            const result = CoreUtils.objectToArray(testObject);
+
+            // Assert
+            expect(result).toContain(testObject.name);
+            expect(result).toContain(testObject.age);
+            expect(result).toHaveLength(2);
+        });
+
+
+        test("given an empty object, return an empty array", () => {
+            // Arrange
+            const testObject = {};
+
+            // Act
+            const result = CoreUtils.objectToArray(testObject);
+
+            // Assert
+            expect(result).toBeEmpty();
+        });
+    });
+
+    // #endregion ObjectToArray
+
+    // -------------------------------------------------------------------------------------------------
+    // #region sleep
+    // -------------------------------------------------------------------------------------------------
+
+    describe("sleep", () => {
+        test("schedule n milliseconds", () => {
+            // Arrange
+            jest.useFakeTimers();
+            const consoleSpy = jest.spyOn(console, "log");
+
+            // Act
+            CoreUtils.sleep(2000, true);
+
+            // At this point in time, only sleep start is printed out
+            expect(consoleSpy).toHaveBeenLastCalledWith("sleep start");
+
+            // Run 1 second later, "sleep end" has not been called yet
+            jest.advanceTimersByTime(1000);
+            expect(consoleSpy).not.toHaveBeenLastCalledWith("sleep end");
+
+            // Fast-forward until all timers have been executed
+            jest.runAllTimers();
+
+            // Assert
+            expect(consoleSpy).toHaveBeenLastCalledWith("sleep end");
+            jest.clearAllTimers();
+            jest.useRealTimers();
+        });
+    });
+
+    // #endregion sleep
+
+    // -------------------------------------------------------------------------------------------------
+    // #region sleepSync
+    // -------------------------------------------------------------------------------------------------
+
+    describe("sleepSync", () => {
+        test("delay n milliseconds", () => {
+            // Arrange && Act
+            const firstCall = performance.now();
+            CoreUtils.sleepSync(3000);
+            const secondCall = performance.now()
+
+            // Assert
+            expect(Math.ceil(secondCall-firstCall)).toBe(3000);
```

Comment:
I don't know if this would always be 3000. I feel that there could be a delay in starting or executing `const secondCall = performance.now()` that would result in 3001+.

Reply:
From what I've tested it always comes close to the number we provided which is pretty strange. I think I can always use toBeGreaterThanOrEqual in this case to account for any delay that can happen between 2 calls

Reply:
Sounds good, I think that is best.